storage/src/main/java/tech/pegasys/artemis/storage/ChainStorageClient.java

```diff
@@ -199,41 +198,53 @@ public void onNewUnprocessedBlock(BeaconBlock block) {
   }
 
   @Subscribe
-  public void onReceievedUnprocessedBlock(Bytes bytes) {
-    BeaconBlock block = BeaconBlock.fromBytes(bytes);
-    onNewUnprocessedBlock(block);
+  public void onReceievedMessage(Bytes bytes) {
+    try {
+      //
+      Attestation attestation = Attestation.fromBytes(bytes);
+      this.eventBus.post(attestation);
+      LOG.log(
+          Level.DEBUG,
+          "Receieved Attestation for block root: "
+              + attestation.getData().getBeacon_block_root().toHexString());
+    } catch (InvalidSSZTypeException e) {
+      BeaconBlock block = BeaconBlock.fromBytes(bytes);
+      this.eventBus.post(block);
+      LOG.log(
+          Level.DEBUG, "Receieved block with state root: " + block.getState_root().toHexString());
+    }
   }
 
   @Subscribe
   public void onNewUnprocessedAttestation(Attestation attestation) {
```

Comment:
@cemozerr can you look at this?

Reply:
Why did you have to comment out the onNewUnprocessedAttestation?